sql/archive.sql

```diff
@@ -0,0 +1,33 @@
+SET statement_timeout=0;
+
+-- Move old contracts to archive tables
+DO $$
+DECLARE
+    contract record;
+    link_contract record;
+BEGIN
+    FOR contract IN SELECT * FROM cards WHERE (type='execute@1.0.0' OR type='session@1.0.0' OR type='external-event@1.0.0' OR active=false)
+        AND created_at < to_char(CURRENT_DATE - INTERVAL '3 months', 'YYYY-MM-DD')::date
+    LOOP
+        RAISE INFO 'Moving contract: [%] [%]', contract.type, contract.id;
+        FOR link_contract IN SELECT * FROM links2 WHERE fromid=contract.id OR toid=contract.id
+        LOOP
+            RAISE INFO 'Moving link: [%]', link_contract.id;
```

Comment:
Whilst looping through 1-by-1 is useful for being able to log each one but in terms of performance it's going to be much slower than writing SQL query/queries that work on the full batch (although precisely what that query/queries will look like depends on the other question)

Reply:
Good point. I was mainly using those output statements for debugging purposes, but the function should be optimized to work on our production table with ~155,000,000 records 😓

Reply:
@Page- Updated the logic to be bit more efficient and not loop one at a time